File: expected_count_TPM_FPKM_data_load.R
##
'
Load Unnormalized Expected Count, TPM, & FPKM Data ~ R version
Rebecca Vislay Wade
01 May 2018

Loads gene expression data for 377 brain samples from the *Aging, Dementia, & TBI
Study* from the Allen Institute for Brain Science. Saves resulting dataframes to
.csv files.

See http://aging.brain-map.org/overview/home for more details about the data.

R 3.4.2
'
##

# load libraries
library(data.table)

# Get TBI_data_files.csv
data_files <- data.frame(fread('http://aging.brain-map.org/data/tbi_data_files.csv'))

# Series of links to normalized FMPK files
data_links <- sample_info['gene_level_fpkm_file_link']

# Grab the list of sample IDs
sample_IDs <- sample_info['rnaseq_profile_id']

# Initialize dataframes
TPM <- data.frame(matrix(nrow = 50283))
FPKM <- data.frame(matrix(nrow = 50283))
raw_read_counts <- data.frame(matrix(nrow = 50283))

# start a timer for DF construction
start <- Sys.time()

# loops through sample files and adds raw read counts ('expected_counts') for each
# sample to a dataframe with the 'rnaseq_profile_id' as header
for (sample in 1:nrow(data_links)){
    print(paste('Sample', sample, 'being added.'))
    flush.console()     # Forces print in loop
    url <- paste('http://aging.brain-map.org', toString(data_links[sample,1]),sep='')
    sample_data <- data.frame(fread(url))
    # Add 'gene_id' from the first file to each dataframe (it's the same for each file)
    if (sample == 1){
        TPM['gene_id'] <- sample_data['gene_id']
        FPKM['gene_id'] <- sample_data['gene_id']
        raw_read_counts['gene_id'] <- sample_data['gene_id']
        }
    TPM[toString(sample_IDs[sample,1])] <- sample_data['TPM']
    FPKM[toString(sample_IDs[sample,1])] <- sample_data['FPKM']
    raw_read_counts[toString(sample_IDs[sample,1])] <- sample_data['expected_count']
    }

# stop timer, calculate running time
stop <- Sys.time()
duration <- stop-start

# drop extra column
TPM <- TPM[,-1]
FPKM <- FPKM[,-1]
raw_read_counts <- raw_read_counts[,-1]

# Write to csv for later
write.csv(raw_read_counts,'raw_read_counts.csv')
write.csv(TPM,'TPM.csv')
write.csv(FPKM,'FPKM.csv')

print(paste('All samples loaded & saved! Loop duration:', round(duration, 2), 'minutes'))

File: ETL/expected_count_TPM_FPKM_data_load.py
##
'''
Load Unnormalized Expected Count, TPM, & FPKM Data ~ Python version
Rebecca Vislay Wade
01 May 2018
25 May 2018 - updated i/o to work with GitHub repo

Loads gene expression data for 377 brain samples from the *Aging, Dementia, & TBI
Study* from the Allen Institute for Brain Science. Saves resulting dataframes to
.csv files.

See http://aging.brain-map.org/overview/home for more details about the data.

Python 3.6.5
'''
##

# Some imports
import pandas as pd
import numpy as np
import time
import os

# load TBI_data_files.csv
data_files = pd.read_csv('http://aging.brain-map.org/data/tbi_data_files.csv')

# Series of links to files containing gene level estimated count/TPM/FPKM for each sample 
data_links = data_files['gene_level_fpkm_file_link']

# Grab RNA-seq sample IDs (these will be the column names in the final DFs)
sample_IDs = data_files['rnaseq_profile_id']

# Initialize DFs
TPM = pd.DataFrame()
FPKM = pd.DataFrame()
raw_read_counts = pd.DataFrame()

# start a timer for dataframe construction
start = time.clock()

# loop through list of links
for sample in range(0, len(data_links)):
    print('Now adding Sample #', sample)
    # get data file for the sample
    url = 'http://aging.brain-map.org' + str(data_links[sample])
    sample_data = pd.read_table(url)
    # get gene_id from the first sample file (they are all the same)
    if sample == 0:
        TPM['gene_id'] = sample_data['gene_id']
        FPKM['gene_id'] = sample_data['gene_id']
        raw_read_counts['gene_id'] = sample_data['gene_id']
    # add TPM, FPKM, and expected_count to respective DataFrames
    TPM[str(sample_IDs[sample])] = sample_data['TPM']
    FPKM[str(sample_IDs[sample])] = sample_data['FPKM']
    raw_read_counts[str(sample_IDs[sample])] = sample_data['expected_count']

# set gene_id as indices
TPM = TPM.set_index('gene_id')
FPKM = FPKM.set_index('gene_id')
raw_read_counts = raw_read_counts.set_index('gene_id')

# stop timer
stop = time.clock()
duration = stop - start

# save to csv for later
os.chdir('..')
TPM.to_csv('data/TPM.csv')
FPKM.to_csv('data/FPKM.csv')
raw_read_counts.to_csv('data/raw_read_counts.csv')

print('All data loaded & saved! Loop duration:', np.round(duration, 2), 'seconds')